doc/nrf/security/crypto/drivers.rst

```diff
@@ -176,24 +178,31 @@ The following table provides an overview of the available directives and their c
    For the complete overview of the available configuration options, see the :ref:`ug_crypto_supported_features` page.
 
 The nRF Security subsystem checks the directives (set through the Kconfig options) to make the optimal driver selection.
-The subsystem combines the ``PSA_WANT_*`` and ``PSA_USE_*`` settings in Kconfig to make the appropriate driver selection for the cryptographic operation:
+The subsystem combines the settings in Kconfig that represent application requirements (``PSA_WANT_*``) and driver capabilities (``PSA_USE_*``) into ``PSA_NEED_*`` macros.
+These macros then control which drivers are compiled into the build and thus available for use at runtime.
+
+* If hardware acceleration is enabled and available for the requested features (as indicated by the relevant ``PSA_NEED_*`` macros), nRF Security selects that driver for performance and security reasons.
+* If no hardware driver is enabled and available, or if the hardware or the driver does not support the specific cryptographic operation (the corresponding ``PSA_NEED_*`` macro is not set for a hardware driver), nRF Security falls back to the software nrf_oberon driver, provided nrf_oberon's ``PSA_NEED_*`` macro is set.
 
-* If hardware acceleration is enabled and available for the requested features, nRF Security selects that driver for performance and security reasons.
-* If no hardware driver is enabled and available, or if the hardware or the driver does not support the specific cryptographic operation, nRF Security falls back to the software nrf_oberon driver.
+.. figure:: ../images/psa_want_use_need_flow.svg
+   :alt: Detailed process for feature and driver selection
+   :align: center
 
 For example:
 
 * You can configure the application to use CC310 and require the AES-192 key size through the :kconfig:option:`CONFIG_PSA_WANT_AES_KEY_SIZE_192` and :kconfig:option:`CONFIG_PSA_USE_CC3XX_KEY_MANAGEMENT_DRIVER` Kconfig options.
+  The resulting ``PSA_NEED_CC3XX_KEY_MANAGEMENT_DRIVER`` macro will be set if both are enabled.
   nRF Security selects the nrf_cc3xx driver if it supports the CCM-AES-192 acceleration with the AES-192 key size.
-  The CC310 peripheral does not support AES keys larger than 128 bits, so nRF Security selects the nrf_oberon driver instead.
-  (This is the case of the Feature A in the previous example.)
+  The CC310 peripheral does not support AES keys larger than 128 bits, so nRF Security selects the nrf_oberon driver instead, if its ``PSA_NEED_OBERON_KEY_MANAGEMENT_DRIVER`` macro is set.
+  (This is the case of the :ref:`Feature A in the example flow <crypto_drivers_example_flow>`.)
 * You can configure the application to require the SHA-512 hashing functionality through the :kconfig:option:`CONFIG_PSA_WANT_ALG_SHA_512` and :kconfig:option:`CONFIG_PSA_USE_CC3XX_HASH_DRIVER` Kconfig options.
+  The resulting ``PSA_NEED_CC3XX_HASH_DRIVER`` macro will be set if both are enabled.
```

Comment:
Same as before, this PSA_NEED will not be enabled if the user only wants SHA512.